### PR TITLE
Add utility predicates to FunctionModel

### PR DESCRIPTION
**Pull request by sauyon**
I seem to remember things like this occurring in more places than this, but I guess I was mistaken. Perhaps I'm just missing something...?

### CODE REVIEW

**Comment by sauyon** [comment]
And this is why I need to reinstate my pre-commit hooks....

File: ql/src/semmle/go/dataflow/internal/DataFlowUtil.qll [hunk 1]
```diff
@@ -896,6 +896,26 @@ class TypeCastNode extends ExprNode {
 abstract class FunctionModel extends Function {
   /** Holds if data flows through this function from `input` to `output`. */
   abstract predicate hasDataFlow(FunctionInput input, FunctionOutput output);
+
+  /** Gets an input node for this model for the call `c`. */
+  DataFlow::Node getInputNode(DataFlow::CallNode c) { this.flowStepForCall(result, _, c) }
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/DataFlowUtil.qll:901]:
```suggestion
  DataFlow::Node getAnInputNode(DataFlow::CallNode c) { this.flowStepForCall(result, _, c) }
```

File: ql/src/semmle/go/dataflow/internal/DataFlowUtil.qll [hunk 2]
```diff
@@ -896,6 +896,26 @@ class TypeCastNode extends ExprNode {
...
+
+  /** Gets an output node for this model for the call `c`. */
+  DataFlow::Node getOutputNode(DataFlow::CallNode c) { this.flowStepForCall(_, result, c) }
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/DataFlowUtil.qll:904]:
```suggestion
  DataFlow::Node getAnOutputNode(DataFlow::CallNode c) { this.flowStepForCall(_, result, c) }
```

File: ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll [hunk 3]
```diff
@@ -140,23 +140,33 @@ predicate sliceStep(DataFlow::Node pred, DataFlow::Node succ) {
...
 /**
  * A model of a function specifying that the function propagates taint from
  * a parameter or qualifier to a result.
  */
 abstract class FunctionModel extends Function {
   /** Holds if taint propagates through this function from `input` to `output`. */
   abstract predicate hasTaintFlow(FunctionInput input, FunctionOutput output);
+
+  /** Gets an input node for this model for the call `c`. */
+  DataFlow::Node getInputNode(DataFlow::CallNode c) { this.taintStepForCall(result, _, c) }
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll:152]:
```suggestion
  DataFlow::Node getAnInputNode(DataFlow::CallNode c) { this.taintStepForCall(result, _, c) }
```

File: ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll [hunk 4]
```diff
@@ -140,23 +140,33 @@ predicate sliceStep(DataFlow::Node pred, DataFlow::Node succ) {
...
+
+  /** Gets an output node for this model for the call `c`. */
+  DataFlow::Node getOutputNode(DataFlow::CallNode c) { this.taintStepForCall(_, result, c) }
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll:155]:
```suggestion
  DataFlow::Node getAnOutputNode(DataFlow::CallNode c) { this.taintStepForCall(_, result, c) }
```

File: ql/src/semmle/go/security/SafeUrlFlowCustomizations.qll [hunk 5]
```diff
@@ -25,6 +25,6 @@ module SafeUrlFlow {
 
   /** A function model step using `UnsafeUrlMethod`, considered as a sanitizer for safe URL flow. */
   private class UnsafeUrlMethodEdge extends SanitizerEdge {
-    UnsafeUrlMethodEdge() { TaintTracking::functionModelStep(any(UnsafeUrlMethod um), this, _) }
+    UnsafeUrlMethodEdge() { this = any(UnsafeUrlMethod um).getInputNode(_) }
```
Comment (smowton) [inline ql/src/semmle/go/security/SafeUrlFlowCustomizations.qll:28]:
```suggestion
    UnsafeUrlMethodEdge() { this = any(UnsafeUrlMethod um).getAnInputNode(_) }
```

**Review by smowton — APPROVED** [review]
Suggest renaming `get[In|Out]putNode` to match their comments, otherwise LGTM!

**Comment by sauyon** [comment]
Changes applied and rebased.

**Pull request merged**